server/models/api.go

```diff
@@ -42,103 +39,116 @@ type Api struct {
 	Annotations        []byte    `datastore:",noindex"` // Serialized annotations.
 }
 
-// NewApiFromParentAndApiID returns an initialized api for a specified parent and ID.
-func NewApiFromParentAndApiID(parent string, id string) (*Api, error) {
-	project, err := names.ParseProject(parent)
+// NewApi initializes a new resource.
+func NewApi(name names.Api, body *rpc.Api) (api *Api, err error) {
+	now := time.Now()
+	api = &Api{
+		ProjectID:          name.ProjectID,
+		ApiID:              name.ApiID,
+		Description:        body.GetDescription(),
+		DisplayName:        body.GetDisplayName(),
+		Availability:       body.GetAvailability(),
+		RecommendedVersion: body.GetRecommendedVersion(),
+		CreateTime:         now,
+		UpdateTime:         now,
+	}
+
+	api.Labels, err = bytesForMap(body.GetLabels())
 	if err != nil {
 		return nil, err
-	} else if err := names.ValidateID(id); err != nil {
-		return nil, err
 	}
 
-	return &Api{
-		ProjectID: project.ProjectID,
-		ApiID:     id,
-	}, nil
-}
-
-// NewApiFromResourceName parses resource names and returns an initialized api.
-func NewApiFromResourceName(name string) (*Api, error) {
-	m, err := names.ParseApi(name)
+	api.Annotations, err = bytesForMap(body.GetAnnotations())
 	if err != nil {
 		return nil, err
 	}
 
-	return &Api{
-		ProjectID: m[1],
-		ApiID:     m[2],
-	}, nil
+	return api, nil
 }
 
 // ResourceName generates the resource name of a api.
-func (api *Api) ResourceName() string {
-	return fmt.Sprintf("projects/%s/apis/%s", api.ProjectID, api.ApiID)
+func (a *Api) ResourceName() string {
```

Comment:
"a *Api" looks weird to me - I know the Go preference for short variable names, but is it required? It seems inconsistent to so often use "api" as the name for *Api variables and to not do it on these method receivers. "api" is already an abbreviation, so this is abbreviating an abbreviation, which (I just think) is overly terse, so how strict is the style guide on this?

Reply:
That's fair. In this case it's less about sticking to short variable names in general and more about receiver names in particular. It's a very common pattern for receiver names to be single characters, so I copied that style here.

Out of curiosity I picked some packages from the standard library. [Most](https://golang.org/pkg/bytes/) [seemed](https://golang.org/pkg/io/) to [stick](https://golang.org/pkg/os/) to [single](https://golang.org/pkg/testing/) [characters](https://golang.org/pkg/time/), but notably I found [regexp](https://golang.org/pkg/regexp/) using `re` and [net/http](https://golang.org/pkg/net/http/) using `srv` and `mux`.

My primary complaint about using `api` for the receiver name is that it reserves that valuable name when the same information (i.e. that we're acting on a receiver) can be conveyed by adopting the common pattern. We can go either way though.